coral/breakpoint/breakpoint_graph.py

```diff
@@ -0,0 +1,1003 @@
+"""Implements a class for BreakpointGraph. Will serve as a container for the
+sequence edges, corcordant edges, discordant edges, and breakpoints inferred
+from a given sequencing file.
+"""
+
+from __future__ import annotations
+
+import logging
+import math
+import time
+import warnings
+from dataclasses import dataclass, field
+from typing import Any
+
+import cvxopt  # type: ignore[import-untyped]
+import cvxopt.modeling  # type: ignore[import-untyped]
+import numpy as np
+
+from coral import types
+from coral.constants import CHR_TAG_TO_IDX
+
+logger = logging.getLogger(__name__)
+
+
+def test_clustering(rc_list, partition, max_multiplicity=5):
```

Comment:
@mattjones315 this function was already renamed - @suhas-r and I suggested ```enumerate_partitions``` or ```enumerate_clustering```

Reply:
For your reference @suhas-r , basically, the goal of ```infer_discordant_edge_multiplicities``` is to infer the "multiplicity", i.e., the number of times a discordant edge can occur in **one ecDNA species**. However, due to the possibility that this discordant edge is shared by different ecDNA species, we actually infer an UPPER LIMIT on the number of times it can occur in any ecDNA species/rearranged genomes (as cycles or paths). Usually an overestimation would have minor impact as the same limit on other discordant edges may still constraint possible complex ecDNA structures, while an underestimation will disallow some true ecDNA structures. The estimation of multiplicities is purely based on read count supporting the discordant edges.